jgrapht-ext/src/test/java/org/jgrapht/ext/DOTImporterTest.java

```diff
@@ -29,6 +29,77 @@
     extends TestCase
 {
 
+    private static class GraphWithID extends AbstractBaseGraph<String,DefaultEdge> implements UndirectedGraph<String,DefaultEdge> {
+
+        String id = null;
+
+        protected GraphWithID() {
+            super(new ClassBasedEdgeFactory<>(DefaultEdge.class), false, false);
+        }
+
+    }
+
+    private static DOTImporter<String, DefaultEdge> idGraphImporter = new DOTImporter<String, DefaultEdge>(null, null,
+            null, (component, attributes) -> {
+        if (component instanceof GraphWithID) {
+            ((GraphWithID) component).id = attributes.getOrDefault("ID", "G");
+        }
+    });
+
+    public void testImportID()
+        throws ImportException
+    {
+        String id = "MyGraph";
+
+        String input = "strict graph " + id + " {\n}\n";
+
+        GraphWithID expected = new GraphWithID();
+        expected.id = id;
+
+        GraphWithID result = new GraphWithID();
+        Assert.assertNull(result.id);
+
+        idGraphImporter.importGraph(result, new StringReader(input));
+        Assert.assertEquals(expected.toString(), result.toString());
+        Assert.assertEquals(expected.id, id);
+
+    }
+
+    public void testImportWrongID()
+            throws ImportException
+    {
+        String invalidID = "2test";
+        String input = "graph " + invalidID + " {\n}\n";
+
+        GraphWithID result = new GraphWithID();
+
+        try {
+            idGraphImporter.importGraph(result, new StringReader(input));
+            Assert.fail("Should not get here");
+        } catch (ImportException e) {
+           Assert.assertEquals(e.getMessage(), "ID in the graph is not formatted correctly: '"
+               + invalidID + "'");
+        }
+    }
+
+    public void testImportOnlyGraphKeyword()
+            throws ImportException
+    {
+        String input = "graph {\n}\n";
+        GraphWithID result = new GraphWithID();
+        idGraphImporter.importGraph(result, new StringReader(input));
+        Assert.assertNull(result.id);
+    }
+
+    public void testImportNoID()
+            throws ImportException
+    {
+        String input = "strict graph {\n}\n";
+        GraphWithID result = new GraphWithID();
+        idGraphImporter.importGraph(result, new StringReader(input));
+        Assert.assertNull(result.id);
+    }
+
```

Comment:
Could you add one more test where the header is malformed? Perhaps having the different parts in the wrong order?